client/packages/coldchain/src/Sensor/api/api.ts

```diff
@@ -27,4 +27,22 @@ export const getSensorQueries = (sdk: Sdk, storeId: string) => ({
         return result?.sensors;
       },
   },
+  update: async (patch: RecordPatch<SensorFragment>) => {
+    const result = await sdk.updateSensor({
+      storeId,
+      input: {
+        id: patch.id,
+        isActive: patch.isActive,
+        name: patch.name,
+        location: patch?.location
+          ? {
+              locationId:
+                patch.location.id == 'null' ? null : patch.location.id,
+            }
+          : undefined,
```

Comment:
I'm not sure if this is the most elegant way to offer graphql the correct structure. Would love to hear if there are better alternatives

Reply:
For context, I've done it this way because the LocationSearchInput uses LocationNodes as the type of the searchable options, and this type has a different locationId structure (and doesn't accept a null value) 

Here I am converting a hard coded 'null' string into null to be passed to the back end in the case that the location of the item or sensor is to be unassigned.

Reply:
sorry I think was a bit missleading with the original suggestion, this is much simpler:

Reply:
https://github.com/openmsupply/open-msupply/compare/2286-sensor-detail-view-suggestion...2286-sensor-detail-view-suggestion(Simplify-generic-method-to-set-nullable-input)

Reply:
Wow yeah awesome, that is great. Thanks Andrei.